### PR TITLE
ING-650: Return fully qualififed index names from query index errs

### PR DESCRIPTION
**Pull request by Westwooo**
Error messages will now be as follows. The IndexNotFound and IndexExists errors still do not contain the fully qualified path as gocbcorex does not know the paths when it gets the error from the server: 

# Index Not Found
```
panic: index not found | {"context":{"details":1,"resource_name":"missingIndex","resource_type":
"queryindex","server":"Query index 'missingIndex' not found."}}
```

# Index Exists
```
panic: index exists | {"context":{"details":1,"resource_name":"testingIndexTwoPointOh",
"resource_type":"queryindex","server":"Query index 'testingIndexTwoPointOh' already existed."}}
```

# Invalid Argument
```
panic: invalid argument | {"context":{"details":1,"resource_name":"test.testing.bucket/testScope/
testCol:tooManyReplicas","resource_type":"queryindex","server":"invalid argument: NumReplicas 
- not enough indexer nodes to create index with replica count"}}
```

# Auth Error
```
panic: server reported that permission to the resource was denied: authentication failure - possible 
reasons - incorrect authentication configuration, bucket doesn’t exist or bucket may be hibernated 
| {"context":{"details":0,"server":"Insufficient permissions to perform query index operation 
against test.testing.bucket/testScope/testCol."}}
```

### CODE REVIEW

File: gateway/dataimpl/server_v1/errorhandler.go [hunk 1]
```diff
@@ -275,22 +275,30 @@ func (e ErrorHandler) NewQueryIndexNotBuildingStatus(baseErr error, bucketName,
...
 }
 
-func (e ErrorHandler) NewQueryIndexAuthenticationFailureStatus(baseErr error, indexName string) *status.Status {
-	st := status.New(codes.PermissionDenied, "Insufficient permissions to perform query index operation.")
-	st = e.tryAttachStatusDetails(st, &epb.ResourceInfo{
-		ResourceType: "queryindex",
-		ResourceName: indexName,
-		Description:  "",
-	})
+func (e ErrorHandler) NewQueryIndexAuthenticationFailureStatus(baseErr error, bucketName, scopeName, collectionName string) *status.Status {
```
[the remark this answers is not in the document]
Comment (Westwooo) [reply gateway/dataimpl/server_v1/errorhandler.go:278]: For details on the AuthFailure error see the gocbcorex PR for ING-650

**Comment by Westwooo** [comment]
> Something worth considering in these cases is that we may want to "enhance" errors being produced by index management calls with the reference when gocbcorex cannot itself tell.

That's what I was thinking about doing, so fully qualifying the index path with the bucket/scope/collection from the callers context. But doesn't that hit the issue where the error being returned could be related to an index other than the one that the index management call is for?

**Comment by brett19** [comment]
In the context of general query error handling, the statement is 'ambiguous' about which scope/index/etc... that the error could be about, since we don't parse the query ourselves, thus it is not safe to be making assumptions in the errors we propagate back to the user.  Especially since in the context of a single query, the query_context can refer to something completely different than the statement does (imagine query_context="bucket1.scope1", statement="SELECT * FROM default:bucket2.scope2").  However, in the context of our index management operations, we actually _do_ know the exact context from the query since we were the ones that generated it, and thus even if the generalized query handling was not able to identify the specific scope/index/etc..., our higher level code that invoked it from the index management could imbue those errors with the additional context they have.

**Comment by brett19** [comment]
I'm not convinced that InvalidArgument needs any path information, but generally looks good.  Avoiding an Approval for now due to the reference to an uncommitted gocbcorex.

**Review by brett19 — DISMISSED** [review]

File: gateway/dataimpl/server_v1/queryserver.go [hunk 2]
```diff
@@ -58,6 +58,15 @@ func (s *QueryServer) translateError(err error) *status.Status {
 	return s.errorHandler.NewGenericStatus(err)
 }
 
+func (s *QueryServer) normalizeDefaultName(name *string) string {
```
[the remark this answers is not in the document]
Comment (Westwooo) [reply gateway/dataimpl/server_v1/queryserver.go:61]: Good point, done!

File: gateway/dataimpl/server_v1/queryserver.go [hunk 3]
```diff
@@ -173,7 +182,11 @@ func (s *QueryServer) Query(in *query_v1.QueryRequest, out query_v1.QueryService
 		result, err = agent.Query(out.Context(), &opts)
 	}
 	if err != nil {
-		return s.translateError(err).Err()
+		return s.translateError(
```
[the remark this answers is not in the document]
Comment (Westwooo) [reply gateway/dataimpl/server_v1/queryserver.go]: My bad, I've changed this now.

**Review by brett19 — APPROVED** [review]

**Pull request merged**